packages/openstef-core/src/openstef_core/feature_engineering/temporal_transforms/cyclic_features.py

```diff
@@ -0,0 +1,228 @@
+        ...     'load': [100, 120, 110]
+        ... }, index=pd.date_range('2025-01-01', periods=3, freq='1h'))
+        >>> dataset = TimeSeriesDataset(data, timedelta(hours=1))
+        >>>
+        >>> # Apply cyclic features with custom configuration
+        >>> config = CyclicFeaturesConfig(included_features=["season", "timeOfDay"])
+        >>> transform = CyclicFeatures(config)
+        >>> transformed = transform.fit_transform(dataset)
+        >>> 'season_sine' in transformed.data.columns
+        True
+        >>> 'time0fday_sine' in transformed.data.columns
+        True
+        >>> 'month_sine' in transformed.data.columns
+        False
+    """
+
+    def __init__(
+        self,
+        config: CyclicFeaturesConfig,
+    ):
+        """Initialize the CyclicFeatures transform with a configuration."""
+        self.config = config
+        self.cyclic_features: pd.DataFrame = pd.DataFrame()
+
+    @staticmethod
+    def _compute_sine(phase: pd.Index, period: float) -> np.ndarray:
+        """Compute sine component for cyclic encoding.
+
+        Args:
+            phase: Position in cycle (e.g., day of year, hour of day).
+            period: Complete cycle length (e.g., 365.25 for yearly).
+
+        Returns:
+            Sine values ranging from -1 to 1.
+        """
+        return np.sin(2 * np.pi * phase / period)
+
+    @staticmethod
+    def _compute_cosine(phase: pd.Index, period: float) -> np.ndarray:
+        """Compute cosine component for cyclic encoding.
+
+        Args:
+            phase: Position in cycle (e.g., day of year, hour of day).
+            period: Complete cycle length (e.g., 365.25 for yearly).
+
+        Returns:
+            Cosine values ranging from -1 to 1.
+        """
+        return np.cos(2 * np.pi * phase / period)
+
+    def _compute_seasonal_feature(self, index: pd.DatetimeIndex) -> pd.DataFrame:
+        """Compute seasonal features based on day of year.
+
+        Args:
+            index: DatetimeIndex to extract day of year from.
+
+        Returns:
+            DataFrame with season_sine and season_cosine columns.
+        """
+        return pd.DataFrame(
+            {
+                "season_sine": self._compute_sine(index.dayofyear, NUM_DAYS_IN_YEAR),
+                "season_cosine": self._compute_cosine(index.dayofyear, NUM_DAYS_IN_YEAR),
+            },
+            index=index,
+        )
+
+    def _compute_monthly_features(self, index: pd.DatetimeIndex) -> pd.DataFrame:
+        """Compute monthly features based on month of year.
+
+        Args:
+            index: DatetimeIndex to extract month from.
+
+        Returns:
+            DataFrame with month_sine and month_cosine columns.
+        """
+        return pd.DataFrame(
+            {
+                "month_sine": self._compute_sine(index.month, NUM_MONTHS_IN_YEAR),
+                "month_cosine": self._compute_cosine(index.month, NUM_MONTHS_IN_YEAR),
+            },
+            index=index,
+        )
```

Comment:
Adding and documenting each of these functions might be a bit overkill. I would suggest creating _create_sine_and_cosine -> pd.DataFrame and then giving them the index in your transform function. Or keeping it since, and cosine, and building the list of pd.Series inside the transform instead.